core/imageboard/image.php

```diff
@@ -292,6 +307,15 @@ private static function terms_to_conditions(array $terms): array
             }
         }
 
+        if ($config->get_bool(HideTagsConfig::ENABLED)) {
+            if (in_array('show_hidden', $terms)) {
+                $key = array_search('show_hidden', $terms);
+                unset($terms[$key]);
+            } else {
+                array_push($terms, '-hidden');
+            }
+        }
+
```

Comment:
I wasn't entirely sure about this implementation... it's a bit strange to "inject" user query terms they never entered... but it _was_ a very easy way to create the feature I wanted 🤔